base/tps/src/org/dogtagpki/server/tps/rest/TokenService.java

```diff
@@ -326,7 +335,14 @@ protected void retrieveTokensWithVLV(
                 throw new PKIException("Token record not found");
             }
 
-            response.addEntry(createTokenData(record));
+            CMS.debug("TokenService.retrieveTokensWithVLV: record.tokenType="+
+ 		    record.getType());
+            if (authorizedProfiles.contains(record.getType())) {
```

Comment:
Can `authorizedProfiles` be null? It's not checked here, but it's checked in line 125.

Reply:
Do we need to check if `authorizedProfiles` contains `ALL_PROFILES`?

Reply:
yes and yes.  As I explained in the patch comment, this is just to demonstrate where and how the permission check is done.  And the main questions I'm asking are
1. is this the right approach? 
2. could this possible be done at ldap search level with a filter that says something like (if the tokenType is one of the authorized profiles)?
thanks!

Reply:
I think a VLV requires a static search filter, but here the filter can change depending on the `authorizedProfiles`, so it might not be possible to do it in LDAP, and this might be the only way to do it.